conan/tools/apple/xcodedeps.py

```diff
@@ -147,6 +147,16 @@ def _merged_vars(name):
             'condition': _xcconfig_conditional(self._conanfile.settings)
         }
 
+        if not require.headers:
```

Comment:
Please not that CMakeDeps is doing:
```python
 if require and not require.libs and not require.headers:
            self.defines = ""
            self.compile_definitions = ""
            self.cxxflags_list = ""
            self.cflags_list = ""
            self.sharedlinkflags_list = ""
            self.exelinkflags_list = ""
            self.objects_list = ""
```
So if there are no headers and libs at all, it is also dropping all definitions and flags.
Maybe we want to discuss this, not sure what is the best approach, but it should be consistent.

Reply:
Yes, I had that code before, but then I realized that when the condition `if not require.libs and not require.headers` is met, no files are generated at all because the dependency is not passed in the loop that iterates over the dependencies, so I thought it made not sense... I'm currently testing that `if not require.libs and not require.headers` happens, no _xcconfig_ files are generated at all.

Reply:
Maybe still necessary to discuss a bit, to consider other cases. The full logic for skipping binaries is:
```python
if require.headers or require.libs or require.run or require.build:
    # do not skip it
```

So build_requires or something that has the ``run=True`` trait, can contribute things. For XcodeDeps, ``build=True`` might not be an issue, but ``run=True`` maybe is? (same for other generators)

Reply:
Yes, then I'll put that code back better and consider the case

Reply:
Done